Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -41,14 +51,16 @@ extension CardNumberViewModel: CardNumberViewModelProtocol {
         // In this case we only update delegate if we have a final result to avoiding
         //    overriding the eager validation with a less informative update
       case let .success((isComplete, scheme)) where isComplete && supportedSchemes.contains(scheme):
-        delegate?.update(cardNumber: cardNumber, scheme: scheme)
+        delegate?.update(result: .success((cardNumber, scheme)))
         return Constants.Bundle.SchemeIcon(scheme: scheme)
       case .success,
            .failure:
+        delegate?.update(result: .failure(.isNotComplete))
         return nil
     }
   }
 
+  // TODO: separate the validation logic from getting the icon
```

Comment:
```suggestion
```

Reply:
same. i will add ticket for refactoring. they are tightly coupled and should have apply SRP/

Reply:
I can understand why this looks tightly coupled, the validation method also reformats the input 👍 . I think I disagree on how this will be improved, but would like a task for it to be created and this TODO removed. TODOs won't get anywhere and just trigger warnings once merged...